langlib/langlib-test/src/test/resources/test-src/xmllib_test.bal

```diff
@@ -1013,3 +1013,117 @@ function testNamespaces() {
     xml:Element element4 = xml:createElement("{http://sample.com/test}bookStore", attributes4, xml ``);
     assertEquals(element4.toString(), "<bookStore xmlns=\"http://sample.com/test\" status=\"online\"/>");
 }
+
+function testIterableOperationsOnUnionType() {
```

Comment:
```ballerina
public function main() {
    xml<xml:Element>|xml<xml:Comment> x1 = xml `<a><b><c><e>foo</e></c></b><d><g>bar</g></d></a>`;
    _ = xml:map(xml:elements(<xml> x1), v => xml:getChildren(v));
}
```
It's better to add a test case similar to the above, where you directly use the cast variable without assigning it to another variable first, and then use it.